src/System.Text.Primitives/System/Text/Parsing/PrimitiveParser2_bool.cs

```diff
@@ -10,6 +10,30 @@ public static partial class PrimitiveParser
     {
         #region helpers
 
+        private static bool IsTrue(ReadOnlySpan<byte> utf8Bytes)
+        {
+            if (utf8Bytes.Length < 4)
+                return false;
+
+            byte firstChar = utf8Bytes[0];
+            if (firstChar != 't' && firstChar != 'T')
```

Comment:
May be a good idea to create small helper function for it that gets always inlined.